ivy_tests/test_ivy/test_frontends/test_tensorflow/test_math.py

```diff
@@ -1382,6 +1382,36 @@ def test_tensorflow_unsorted_segment_sqrt_n(
     )
 
 
+# unsorted_segment_prod
+@handle_frontend_test(
+    fn_tree="tensorflow.math.unsorted_segment_prod",
+    data=helpers.array_values(dtype=ivy.int32, shape=(5, 6), min_value=1, max_value=9),
```

Comment:
We should be testing for all supported cases, so it would be best to not limit the `shape` and to sample the `dtype` from all "valid" dtypes. 
I would suggest having a look at how other tests generate the input arrays.